Sources/SwiftParser/Specifiers.swift

```diff
@@ -795,22 +796,56 @@ extension Parser {
       ) {
         let misspelledThrows = self.eat(handle)
         unexpected.append(misspelledThrows)
-        if effectSpecifiers?.throwsClause == nil {
-          if matchedSubset == S.CorrectThrowsTokenKinds.self {
-            synthesizedThrows = missingToken(spec)
-          } else {
-            synthesizedThrows = missingToken(.throws)
+        guard effectSpecifiers?.throwsClause == nil else {
+          continue
+        }
+        if matchedSubset == S.CorrectThrowsTokenKinds.self {
+          let synthesizedThrows = missingToken(spec)
+          var lookahead = self.lookahead()
+          guard lookahead.consume(if: .leftParen) != nil,
```

Comment:
Could you add a check for whether the is a space between `throws` and the left parenthesis, similar to here

https://github.com/swiftlang/swift-syntax/blob/93df9ed82e07760eb8eceb09a4504d7057110da3/Sources/SwiftParser/Attributes.swift#L184

A test case for this would be that `func foo() -> throws (Int, Int)` has a Fix-It to move the `throws` but not `(Int, Int)`.